src/analyses/goto_check.cpp

```diff
@@ -1902,32 +1913,64 @@ void goto_checkt::goto_check(
-        flag_resetter.set_flag(enable_pointer_overflow_check, false);
+        seen = flag_resetter.set_flag(enable_pointer_overflow_check, false);
       else if(d.first == "disable:float-overflow-check")
-        flag_resetter.set_flag(enable_float_overflow_check, false);
+        seen = flag_resetter.set_flag(enable_float_overflow_check, false);
       else if(d.first == "disable:conversion-check")
-        flag_resetter.set_flag(enable_conversion_check, false);
+        seen = flag_resetter.set_flag(enable_conversion_check, false);
       else if(d.first == "disable:undefined-shift-check")
-        flag_resetter.set_flag(enable_undefined_shift_check, false);
+        seen = flag_resetter.set_flag(enable_undefined_shift_check, false);
       else if(d.first == "disable:nan-check")
-        flag_resetter.set_flag(enable_nan_check, false);
+        seen = flag_resetter.set_flag(enable_nan_check, false);
       else if(d.first == "disable:pointer-primitive-check")
-        flag_resetter.set_flag(enable_pointer_primitive_check, false);
+        seen = flag_resetter.set_flag(enable_pointer_primitive_check, false);
+      else if(d.first == "enable:bounds-check")
+        seen = flag_resetter.set_flag(enable_bounds_check, true);
+      else if(d.first == "enable:pointer-check")
+        seen = flag_resetter.set_flag(enable_pointer_check, true);
+      else if(d.first == "enable:memory-leak-check")
+        seen = flag_resetter.set_flag(enable_memory_leak_check, true);
+      else if(d.first == "enable:div-by-zero-check")
+        seen = flag_resetter.set_flag(enable_div_by_zero_check, true);
+      else if(d.first == "enable:enum-range-check")
+        seen = flag_resetter.set_flag(enable_enum_range_check, true);
+      else if(d.first == "enable:signed-overflow-check")
+        seen = flag_resetter.set_flag(enable_signed_overflow_check, true);
+      else if(d.first == "enable:unsigned-overflow-check")
+        seen = flag_resetter.set_flag(enable_unsigned_overflow_check, true);
+      else if(d.first == "enable:pointer-overflow-check")
+        seen = flag_resetter.set_flag(enable_pointer_overflow_check, true);
+      else if(d.first == "enable:float-overflow-check")
+        seen = flag_resetter.set_flag(enable_float_overflow_check, true);
+      else if(d.first == "enable:conversion-check")
+        seen = flag_resetter.set_flag(enable_conversion_check, true);
+      else if(d.first == "enable:undefined-shift-check")
+        seen = flag_resetter.set_flag(enable_undefined_shift_check, true);
+      else if(d.first == "enable:nan-check")
+        seen = flag_resetter.set_flag(enable_nan_check, true);
+      else if(d.first == "enable:pointer-primitive-check")
+        seen = flag_resetter.set_flag(enable_pointer_primitive_check, true);
+
+      INVARIANT(
+        !seen,
+        "Found enable and disable pragmas for " + id2string(d.first) +
+          " at \n" + i.source_location().pretty());
```

Comment:
Shouldn't this invariant just be maintained within `set_flag`? It seems it is checked every time you use `set_flag`.

Reply:
`flag_resett` does not know about flag names and instructions so we would not be able to generate a readable error message without push lots of information inside `flag_resett`. I could capture context using a lambda and pass it into set_flag but it would not seem cumbersome.